File: TP01/TP01Q06.java
class TP01Q06 {

    
    public static void main(String[] args) 
    {
        String linha = "";
        linha = MyIO.readLine();
        while (isFim(linha) == false)
        {
            if (isVowels(linha) == true)
            {
                System.out.print("SIM ");
            }
            else
            {
                System.out.print("NAO ");
            }
            
            if (isConsonants(linha) == true)
            {
                System.out.print("SIM ");
            }
            else
            {
                System.out.print("NAO ");
            }

            if (isNumbers(linha) == true)
            {
                System.out.print("SIM ");
            }
            else
            {
                System.out.print("NAO ");
            }

            if (isRoyalNumbers(linha) == true)
            {
                System.out.println("SIM ");
            }
            else
            {
                System.out.println("NAO ");
            }
            linha = MyIO.readLine();
        }
    }

    public static boolean isFim(String str)
    {
        return(str.length() == 3 && str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M');
    }

    public static boolean isVowels(String linha) 
    {
        boolean checkStatus = true;
        for (int i = 0; i < linha.length(); i++) 
        {
            if (linha.charAt(i) == 'a' || linha.charAt(i) == 'e' || linha.charAt(i) == 'i' || linha.charAt(i) == 'o'|| linha.charAt(i) == 'u' || linha.charAt(i) == 'A' || linha.charAt(i) == 'E'|| linha.charAt(i) == 'I' || linha.charAt(i) == 'O' || linha.charAt(i) == 'U') 
            {
                checkStatus = true;
            } else 
            {
                checkStatus = false;
                return checkStatus;
            }
        }

        return checkStatus;
    }

    public static boolean isConsonants(String linha) 
    {
        boolean checkStatus = true;
        for (int i = 0; i < linha.length(); i++) {
            if (linha.charAt(i) == 'a' || linha.charAt(i) == 'e' || linha.charAt(i) == 'i' || linha.charAt(i) == 'o'|| linha.charAt(i) == 'u' || linha.charAt(i) == 'A' || linha.charAt(i) == 'E'|| linha.charAt(i) == 'I' || linha.charAt(i) == 'O' || linha.charAt(i) == 'U') 
            {
                checkStatus = false;
                return checkStatus;
            } else 
            {
                checkStatus = true;

            }
        }

        return checkStatus;
    }

    public static boolean isNumbers(String linha) 
    {
        boolean checkStatus = true;
        for (int i=0;i<linha.length();i++)
        {
            if((linha.charAt(i) >= '0' && linha.charAt(i) <= '9'))
            {
                checkStatus=true;
            }
            else
            {
                checkStatus=false;
                return checkStatus;
            }
        }
        return checkStatus;
    }

    public static boolean isRoyalNumbers(String linha) 
    {
        boolean checkStatus = true;
        for (int i=0;i<linha.length();i++)
        {
            if((linha.charAt(i) >= '0' && linha.charAt(i) <= '9')||(linha.charAt(i) == ',' || linha.charAt(i) == '.'))
            {
                checkStatus=true;
            }
            else
            {
                checkStatus=false;
                return checkStatus;
            }
        }
        return checkStatus;

    }

    public static boolean isLetra(char x)
    {
         boolean ret1 = false;
        if(x >= 'A' && x <= 'Z' || x >= 'a' && x <= 'z')
            ret1 = true;

        return ret1;
    }




}

File: TP01/TP01Q06.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <stdbool.h>
#include <string.h>

bool isFim(char *str)
{

    if ((str[0] == 'F') && (str[1] == 'I') && (str[2] == 'M'))
    {
        return true;
    }
    else
    {
        return false;
    }
}

bool isNumbers(char *linha) 
    {
        bool checkStatus = true;
        for (int i=0;i<strlen(linha);i++)
        {
            if((linha[i] >= '0' && linha[i] <= '9'))
            {
                checkStatus=true;
            }
            else
            {
                checkStatus=false;
                return checkStatus;
            }
        }
        return checkStatus;
    }

bool isVowels(char *linha) 
    {
        bool checkStatus = true;
        for (int i = 0; i < strlen(linha); i++) 
        {
            if (linha[i] == 'a' || linha[i] == 'e' || linha[i] == 'i' || linha[i] == 'o'|| linha[i] == 'u' || linha[i] == 'A' || linha[i] == 'E'|| linha[i] == 'I' || linha[i] == 'O' || linha[i] == 'U') 
            {
                checkStatus = true;
            } else 
            {
                checkStatus = false;
                return checkStatus;
            }
        }

        return checkStatus;
    }

bool isLetra(char x)
{
    bool ret1 = false;
    if (x >= 'A' && x <= 'Z' || x >= 'a' && x <= 'z')
        ret1 = true;

    return ret1;
}

bool isRoyalNumbers(char *linha)
{
    bool checkStatus = true;
    for (int i = 0; i < strlen(linha); i++)
    {
        if ((linha[i] >= '0' && linha[i] <= '9') || (linha[i] == ',' || linha[i] == '.'))
        {
            checkStatus = true;
        }
        else
        {
            checkStatus = false;
            return checkStatus;
        }
    }
    return checkStatus;
}

bool isConsonants(char *linha)
{
    bool checkStatus = true;
    for (int i = 0; i < strlen(linha); i++)
    {
        if (linha[i] == 'a' || linha[i] == 'e' || linha[i] == 'i' || linha[i] == 'o' || linha[i] == 'u' || linha[i] == 'A' || linha[i] == 'E' || linha[i] == 'I' || linha[i] == 'O' || linha[i] == 'U')
        {
            checkStatus = false;
            return checkStatus;
        }
        else
        {
            checkStatus = true;
        }
    }
}


int main(void)
{
    char linha[999] = "";
    scanf(" %[^\n]s", linha);
    while (isFim(linha) == false)
        {
            if (isVowels(linha) == true)
            {
                printf("SIM ");
            }
            else
            {
                printf("NAO ");
            }
            
            if (isConsonants(linha) == true)
            {
                printf("SIM ");
            }
            else
            {
                printf("NAO ");
            }

            if (isNumbers(linha) == true)
            {
                printf("SIM ");
            }
            else
            {
                printf("NAO ");
            }

            if (isRoyalNumbers(linha) == true)
            {
                printf ("SIM\n");
            }
            else
            {
                printf ("NAO\n");
            }
            scanf(" %[^\n]s", linha);
        }
    return 0;
}